app/src/main/java/com/github/emilehreich/bootcamp/ApiActivity.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.emilehreich.bootcamp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.AndroidEntryPoint
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.HttpException
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class ApiActivity : AppCompatActivity() {
```

Comment:
For this implementation, maybe it would have been better to use a model-view framework to help you test your code